### DISCUSSION TITLE
How do I transfer my Contribution activity to the newly added email address?

### DISCUSSION THREAD

**Discussion by Dollarsssss** [community/community#141930] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I want to transfer because my old email was hacked.

**Comment by mikk1max** [comment]
### Try these steps:

1. Add the New Email to Your GitHub Account
2. Update the Email on Your Past Commits:
```bash
pip install git-filter-repo
```
```bash
git clone https://github.com/your-username/your-repo.git
cd your-repo
```
```bash
git filter-repo --email-callback '
if commit.author_email == b"old-email@example.com":
    commit.author_email = b"new-email@example.com"
    commit.committer_email = b"new-email@example.com"
'
```
```bash
git push --force --all
```
GitHub identifies contributions based on the email address used in the commit, so as long as the new email is added and verified, your contributions should transfer. Since your old email was hacked, make sure you remove it from your GitHub account under the email settings after ensuring the contributions are linked to the new email.
